userapi/src/main/java/fk/prof/userapi/Configuration.java

```diff
@@ -0,0 +1,202 @@
+package fk.prof.userapi;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.json.JsonObject;
+
+import javax.validation.Valid;
+import javax.validation.constraints.AssertTrue;
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Created by gaurav.ashok on 15/05/17.
+ */
+public class Configuration {
+
+  @JsonProperty("vertxOptions")
+  private VertxOptions vertxOptions = new VertxOptions();
+
+  @JsonProperty("profile.retention.duration.min")
+  private Integer profileRetentionDurationMin = 30;
+
+  @JsonProperty("aggregation_window.duration.secs")
+  private Integer aggregationWindowDurationSec = 1800;
+
+  @NotNull
+  @JsonProperty("userapiHttpOptions")
+  private DeploymentOptions httpVerticleConfig;
+
+  @NotNull
+  @Valid
+  @JsonIgnore
+  private HttpConfig httpConfig;
+
+  @NotNull
+  @Valid
+  @JsonProperty("storage")
+  private StorageConfig storageConfig;
+
+  @NotNull
+  @JsonProperty("base.dir")
+  private String baseDir;
+
+  public VertxOptions getVertxOptions() {
+    return vertxOptions;
+  }
+
+  public Integer getProfileRetentionDurationMin() {
+    return profileRetentionDurationMin;
+  }
+
+  public Integer getAggregationWindowDurationSec() {
+    return aggregationWindowDurationSec;
+  }
+
+  public DeploymentOptions getHttpVerticleConfig() {
+    return httpVerticleConfig;
+  }
+
+  public HttpConfig getHttpConfig() {
+    return httpConfig;
+  }
+
+  public StorageConfig getStorageConfig() {
+    return storageConfig;
+  }
+
+  public String getBaseDir() {
+    return baseDir;
+  }
+
+  private void setVertxOptions(Map<String, Object> vertxOptionsMap) {
+    this.vertxOptions = new VertxOptions(new JsonObject(vertxOptionsMap));
+  }
+
+  private void setHttpVerticleConfig(Map<String, Object> httpVerticleConfigMap) {
+    httpVerticleConfig = new DeploymentOptions(new JsonObject(httpVerticleConfigMap));
+    httpConfig = httpVerticleConfig.getConfig().mapTo(HttpConfig.class);
+  }
+
+  public static class HttpConfig {
```

Comment:
Since we are refactoring the configs, we can push for standardising config structure (will help us reuse these generic pieces e.g httpConfig, storageConfig, ThreadPool config) for backend as well.